transport/src/androidTest/kotlin/com/genesys/cloud/messenger/transport/core/messagingclient/MCMessageTests.kt

```diff
@@ -84,5 +93,32 @@ class MCMessageTests : BaseMessagingClientTest() {
                 "Message rate too high for this session. Retry after 3 seconds."
             )
         }
+        verify(exactly = 0) {
+            mockCustomAttributesStore.onError()
+        }
+    }
+
+    @Test
+    fun `when SocketListener invoke onMessage with CustomAttributesTooLarge error message`() {
+        val expectedErrorMessage = "Custom Attributes in channel metadata is larger than 2048 bytes"
+        subject.connect()
+
+        slot.captured.onMessage(Response.customAttributeSizeTooLarge)
+
+        verifySequence {
+            connectSequence()
+            mockCustomAttributesStore.onError()
+            mockMessageStore.onMessageError(
+                ErrorCode.CustomAttributeSizeTooLarge,
+                expectedErrorMessage
+            )
+            mockAttachmentHandler.onMessageError(
```

Comment:
so in case of CA failure on message sending, the message should be resent by the SDK consumer?
You are not resending the message without the CA, if failed ?

Reply:
No, because potentially, there can be multiple reasons for the failure.